depthai_sdk/src/depthai_sdk/managers/encoding_manager.py

```diff
@@ -48,16 +50,24 @@ def createDefaultQueues(self, device):
 
         for cameraName, node in self._encodingNodes.items():
             self._encodingQueues[cameraName] = device.getOutputQueue(cameraName + "EncXout", maxSize=30, blocking=True)
-            suffix = ".h265" if node.getProfile() == dai.VideoEncoderProperties.Profile.H265_MAIN else ".h264"
-            self._encodingFiles[cameraName] = (self.encodeOutput / cameraName).with_suffix(suffix).open('wb')
+            # suffix = ".h265" if node.getProfile() == dai.VideoEncoderProperties.Profile.H265_MAIN else ".h264"
+            if self.encodeConfig[cameraName][1] == ".mcap":
+                self._mcap[cameraName] = DepthAiMcap(cameraName)
+                self._mcap[cameraName].imageInit(cameraName)
+            else:
+                self._encodingFiles[cameraName] = (self.encodeOutput / cameraName).with_suffix(self.encodeConfig[cameraName][1]).open('wb')
 
     def parseQueues(self):
         """
         Parse the output queues, consuming the available data packets in it and storing them inside opened stream files
         """
         for name, queue in self._encodingQueues.items():
             while queue.has():
-                queue.get().getData().tofile(self._encodingFiles[name])
+                if self.encodeConfig[name][1] == ".mcap":
+                    self._mcap[name].imageSave(queue.get().getData(), name)
```

Comment:
this should be a `value` of `EncodingTypes` enum (which is a class currently). 
It should behave the same as `Previews` enum - where we use `Preview.color.name` to get a stream name and `Preview.color.value(packet, <other_args>)` to get an actual frame.

Same should apply here, each encoding type will have it's own method for handling received packet and save it to the disk